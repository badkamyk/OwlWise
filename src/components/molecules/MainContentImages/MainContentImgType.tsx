export type MainContentImagesType = {
	firstSrc: string;
	firstAlt: string;
	secondSrc: string;
	secondAlt: string;
	firstClass?: string;
	secondClass?: string;
};
