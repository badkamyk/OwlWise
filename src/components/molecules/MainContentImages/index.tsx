import MainCardImg from "../../atoms/MainCardImg";
import { MainContentImagesType } from "./MainContentImgType";

export default function MainContentImages({
	firstSrc,
	firstAlt,
	secondSrc,
	secondAlt,
	firstClass,
	secondClass,
}: MainContentImagesType) {
	return (
		<div className="mt-8 grid grid-cols-2 gap-4">
			<MainCardImg src={firstSrc} alt={firstAlt} className={firstClass} />
			<MainCardImg src={secondSrc} alt={secondAlt} className={secondClass} />
		</div>
	);
}
