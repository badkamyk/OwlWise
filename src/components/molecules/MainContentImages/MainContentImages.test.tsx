import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import MainContentImages from "./index";

it("renders MainContentImages with props correctly", () => {
	const { container } = render(
		<MainContentImages
			firstSrc={"https://via.placeholder.com/150"}
			firstAlt={"firstAlt"}
			secondSrc={"https://via.placeholder.com/150"}
			secondAlt={"secondAlt"}
			firstClass={"firstClass"}
			secondClass={"secondClass"}
		/>
	);
	const allImages = screen.getAllByRole("img");
	expect(
		container.getElementsByClassName("mt-8 grid grid-cols-2 gap-4")
	).toBeTruthy();
	expect(allImages).toHaveLength(2);
	allImages.forEach((image, i) => {
		expect(image).toHaveProperty("src", "https://via.placeholder.com/150");
		expect(image).toHaveProperty("alt", i === 0 ? "firstAlt" : "secondAlt");
	});
	expect(container).toMatchSnapshot();
});
