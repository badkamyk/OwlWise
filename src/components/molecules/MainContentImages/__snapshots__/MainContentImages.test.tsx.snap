// Vitest Snapshot v1

exports[`MainContentImages 1`] = `
<div>
  <div
    class="mt-8 grid grid-cols-2 gap-4"
  >
    <img
      alt="firstAlt"
      class="firstClass"
      src="https://via.placeholder.com/150"
    />
    <img
      alt="secondAlt"
      class="secondClass"
      src="https://via.placeholder.com/150"
    />
  </div>
</div>
`;

exports[`renders MainContentImages with props correctly 1`] = `
<div>
  <div
    class="mt-8 grid grid-cols-2 gap-4"
  >
    <img
      alt="firstAlt"
      class="firstClass"
      src="https://via.placeholder.com/150"
    />
    <img
      alt="secondAlt"
      class="secondClass"
      src="https://via.placeholder.com/150"
    />
  </div>
</div>
`;
